`opam install alt-ergo-lib` fails to find opam (nonstandard $HOME)

I'm trying to install `alt-ergo` on an institute machine where `$HOME` is set to `/user/cdoczkal/home`, which is a symlink to `/home/cdoczkal`. The `opam` binary is installed to `~/.local/bin/opam`, which is included in `$PATH`. However, when I invoke`opam install -vv alt-ergo` I get the following error:
 ```
- which: no opam in (/home/cdoczkal/.opam/default/bin:/user/cdoczkal/home/.local/bin:/user/cdoczkal/home/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin)
- ERROR: Couldn't find opam in env.
```

Since I could not install the packages from the official repository, I downloaded the sources and pinned the included `.opam` files. However, even in this setup, the configure script reports that it cannot find `opam`. 

As a workaround, I replaced the first line of the build instructions as follows:
```
  # ["ocaml" "unix.cma" "configure.ml" name "--libdir" lib "--mandir" man]
  ["./configure" "--prefix" prefix name] 
```
By calling `configure` with `--prefix`, the package installs just fine.

Full error trace below:

```
doczkal@doczkal alt-ergo-2.4.0 $ opam install -vv alt-ergo-lib

<><> Synchronising pinned packages ><><><><><><><><><><><><><><><><><><><><><><>
+ /usr/bin/rsync "-rLptgoDrvc" "--exclude" ".git" "--exclude" "_darcs" "--exclude" ".hg" "--exclude" ".#*" "--exclude" "_opam*" "--delete" "--delete-excluded" "/home/cdoczkal/src-compile/alt-ergo-2.4.0/" "/home/cdoczkal/.opam/default/.opam-switch/sources/alt-ergo-lib"
- sending incremental file list
- 
- sent 40,865 bytes  received 107 bytes  81,944.00 bytes/sec
- total size is 4,946,071  speedup is 120.72
[alt-ergo-lib.2.4.0] no changes from file:///home/cdoczkal/src-compile/alt-ergo-2.4.0

The following actions will be performed:
  ∗ install alt-ergo-lib 2.4.0*

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+ /home/cdoczkal/.opam/opam-init/hooks/sandbox.sh "build" "ocaml" "unix.cma" "configure.ml" "alt-ergo-lib" "--libdir" "/home/cdoczkal/.opam/default/lib" "--mandir" "/home/cdoczkal/.opam/default/man" (CWD=/home/cdoczkal/.opam/default/.opam-switch/build/alt-ergo-lib.2.4.0)
- which: no opam in (/home/cdoczkal/.opam/default/bin:/user/cdoczkal/home/.local/bin:/user/cdoczkal/home/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin)
- ERROR: Couldn't find opam in env.
- To solve this, you can either install opam, provide an explicit value to the
- `--prefix` argument of this script, or provide explicit values to both the
- `--libdir` and `--mandir` options of this
- script.
[ERROR] The compilation of alt-ergo-lib failed at "/home/cdoczkal/.opam/opam-init/hooks/sandbox.sh build ocaml unix.cma configure.ml alt-ergo-lib --libdir /home/cdoczkal/.opam/default/lib --mandir /home/cdoczkal/.opam/default/man".

#=== ERROR while compiling alt-ergo-lib.2.4.0 =================================#
# context     2.0.8 | linux/x86_64 | ocaml-base-compiler.4.12.0 | pinned(file:///home/cdoczkal/src-compile/alt-ergo-2.4.0)
# path        /home/cdoczkal/.opam/default/.opam-switch/build/alt-ergo-lib.2.4.0
# command     /home/cdoczkal/.opam/opam-init/hooks/sandbox.sh build ocaml unix.cma configure.ml alt-ergo-lib --libdir /home/cdoczkal/.opam/default/lib --mandir /home/cdoczkal/.opam/default/man
# exit-code   1
# env-file    /home/cdoczkal/.opam/log/alt-ergo-lib-50591-05296d.env
# output-file /home/cdoczkal/.opam/log/alt-ergo-lib-50591-05296d.out
### output ###
# which: no opam in (/home/cdoczkal/.opam/default/bin:/user/cdoczkal/home/.local/bin:/user/cdoczkal/home/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin)
# ERROR: Couldn't find opam in env.
# To solve this, you can either install opam, provide an explicit value to the
# `--prefix` argument of this script, or provide explicit values to both the
# `--libdir` and `--mandir` options of this
# script.



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build alt-ergo-lib 2.4.0
└─ 
╶─ No changes have been performed
'opam install -vv alt-ergo-lib' failed.
cdoczkal@doczkal alt-ergo-2.4.0 $ which opam
~/.local/bin/opam
cdoczkal@doczkal alt-ergo-2.4.0 $ echo $PATH
/home/cdoczkal/.opam/default/bin:/user/cdoczkal/home/.local/bin:/user/cdoczkal/home/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin
cdoczkal@doczkal alt-ergo-2.4.0 $ echo $HOME
/user/cdoczkal/home
cdoczkal@doczkal alt-ergo-2.4.0 $ ls -l /user/cdoczkal/
total 12
lrwxrwxrwx 1 root root   14 Dec  2  2019 home -> /home/cdoczkal
drwxr-xr-x 2 root root 4096 Dec  2  2019 laptop-doczkal
drwxr-xr-x 2 root root 4096 Nov 28  2019 nas
drwxr-xr-x 2 root root 4096 Nov 28  2019 ses
cdoczkal@doczkal alt-ergo-2.4.0 $ 

```

--- a/alt-ergo-lib.opam
+++ b/alt-ergo-lib.opam
@@ -33,7 +33,7 @@ license: [
 ]
 
 build: [
-  ["ocaml" "unix.cma" "configure.ml" name "--libdir" lib "--mandir" man]
+  ["ocaml" "unix.cma" "configure.ml" name "--prefix" prefix "--libdir" lib "--mandir" man]
   ["dune" "subst"] {pinned}
   ["dune" "build" "-p" name "-j" jobs]
 ]

--- a/alt-ergo-lib.opam.template
+++ b/alt-ergo-lib.opam.template
@@ -6,7 +6,7 @@ license: [
 ]
 
 build: [
-  ["ocaml" "unix.cma" "configure.ml" name "--libdir" lib "--mandir" man]
+  ["ocaml" "unix.cma" "configure.ml" name "--prefix" prefix "--libdir" lib "--mandir" man]
   ["dune" "subst"] {pinned}
   ["dune" "build" "-p" name "-j" jobs]
 ]

--- a/alt-ergo-parsers.opam
+++ b/alt-ergo-parsers.opam
@@ -32,7 +32,7 @@ license: [
 ]
 
 build: [
-  ["ocaml" "unix.cma" "configure.ml" name "--libdir" lib "--mandir" man]
+  ["ocaml" "unix.cma" "configure.ml" name "--prefix" prefix "--libdir" lib "--mandir" man]
   ["dune" "subst"] {pinned}
   ["dune" "build" "-p" name "-j" jobs]
 ]

--- a/alt-ergo-parsers.opam.template
+++ b/alt-ergo-parsers.opam.template
@@ -6,7 +6,7 @@ license: [
 ]
 
 build: [
-  ["ocaml" "unix.cma" "configure.ml" name "--libdir" lib "--mandir" man]
+  ["ocaml" "unix.cma" "configure.ml" name "--prefix" prefix "--libdir" lib "--mandir" man]
   ["dune" "subst"] {pinned}
   ["dune" "build" "-p" name "-j" jobs]
 ]

--- a/alt-ergo.opam
+++ b/alt-ergo.opam
@@ -30,7 +30,7 @@ license: [
 ]
 
 build: [
-  ["ocaml" "unix.cma" "configure.ml" name "--libdir" lib "--mandir" man]
+  ["ocaml" "unix.cma" "configure.ml" name "--prefix" prefix "--libdir" lib "--mandir" man]
   ["dune" "subst"] {pinned}
   ["dune" "build" "-p" name "-j" jobs]
 ]

--- a/alt-ergo.opam.template
+++ b/alt-ergo.opam.template
@@ -7,7 +7,7 @@ license: [
 ]
 
 build: [
-  ["ocaml" "unix.cma" "configure.ml" name "--libdir" lib "--mandir" man]
+  ["ocaml" "unix.cma" "configure.ml" name "--prefix" prefix "--libdir" lib "--mandir" man]
   ["dune" "subst"] {pinned}
   ["dune" "build" "-p" name "-j" jobs]
 ]

--- a/altgr-ergo.opam
+++ b/altgr-ergo.opam
@@ -32,7 +32,7 @@ license: [
 ]
 
 build: [
-  ["ocaml" "unix.cma" "configure.ml" name "--libdir" lib "--mandir" man]
+  ["ocaml" "unix.cma" "configure.ml" name "--prefix" prefix "--libdir" lib "--mandir" man]
   ["dune" "subst"] {pinned}
   ["dune" "build" "-p" name "-j" jobs]
 ]

--- a/altgr-ergo.opam.template
+++ b/altgr-ergo.opam.template
@@ -6,7 +6,7 @@ license: [
 ]
 
 build: [
-  ["ocaml" "unix.cma" "configure.ml" name "--libdir" lib "--mandir" man]
+  ["ocaml" "unix.cma" "configure.ml" name "--prefix" prefix "--libdir" lib "--mandir" man]
   ["dune" "subst"] {pinned}
   ["dune" "build" "-p" name "-j" jobs]
 ]

